modules/lambda-registrator/variables.tf

```diff
@@ -32,20 +30,20 @@ variable "enterprise" {
   default     = false
 }
 
-variable "partitions" {
-  description = "Specifies the partitions that Lambda registrator will manage [Consul Enterprise]."
+variable "admin_partitions" {
+  description = "Specifies the admin partitions that Lambda function registrator will manage [Consul Enterprise]."
   type        = list(string)
-  default     = []
+  default     = var.enterprise ? [] : null
```

Comment:
We try to avoid using `null` as a default value, because the Terraform registry docs will show it as a required field if the the default value is null.

Reply:
Ahh. Rock and a hard place here 😂 Seems like optionals would be nice here, hopefully those will become more than experimental in the future. 👍

Reply:
My understanding is that we are to call the feature "admin partitions" but to use the word "partition" in all flags/config. See https://www.consul.io/docs/agent/config/config-files#partition-1, https://www.consul.io/docs/agent/config/config-files#partition